src/3d/framegraph/qgsforwardrenderview.h

```diff
@@ -0,0 +1,173 @@
+    // QDebugOverlay added in the forward pass
+#if QT_VERSION >= QT_VERSION_CHECK( 5, 15, 0 )
+    Qt3DRender::QDebugOverlay *mDebugOverlay = nullptr;
+#endif
+
+    /**
+     * We define three forward passes: one for solid objects, followed by two for transparent objects (one to write colors but no depths, one to write depths) :
+     *
+     *                                  |
+     *                         +-----------------+
+     *                         | QCameraSelector |  (using the main camera)
+     *                         +-----------------+
+     *                                  |
+     *                         +-----------------+
+     *                         |  QLayerFilter   |  (using mForwardRenderLayer)
+     *                         +-----------------+
+     *                                  |
+     *                         +-----------------+
+     *                         | QRenderStateSet |  define clip planes
+     *                         +-----------------+
+     *                                  |
+     *                      +-----------------------+
+     *                      | QRenderTargetSelector | (write mForwardColorTexture + mForwardDepthTexture)
+     *                      +-----------------------+
+     *                                  |
+     *         +------------------------+---------------------+
+     *         |                                              |
+     *  +-----------------+    discard               +-----------------+    accept
+     *  |  QLayerFilter   |  transparent             |  QLayerFilter   |  transparent
+     *  +-----------------+    objects               +-----------------+    objects
+     *         |                                              |
+     *  +-----------------+  use depth test          +-----------------+   sort entities
+     *  | QRenderStateSet |  cull back faces         |  QSortPolicy    |  back to front
+     *  +-----------------+                          +-----------------+
+     *         |                                              |
+     *  +-----------------+              +--------------------+--------------------+
+     *  | QFrustumCulling |              |                                         |
+     *  +-----------------+     +-----------------+  use depth tests      +-----------------+  use depth tests
+     *         |                | QRenderStateSet |  don't write depths   | QRenderStateSet |  write depths
+     *         |                +-----------------+  write colors         +-----------------+  don't write colors
+     *  +-----------------+                          use alpha blending                        don't use alpha blending
+     *  |  QClearBuffers  |  color and depth         no culling                                no culling
+     *  +-----------------+
+     *         |
+     *  +-----------------+
+     *  |  QDebugOverlay  |
+     *  +-----------------+
```

Comment:
a misunderstanding probably - sorry if I was unclear - I meant that this diagram should go to the top of buildRenderPasses() function's body, so that it is together with the code that actually builds the pass...

Reply:
ok